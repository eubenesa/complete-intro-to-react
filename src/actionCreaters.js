// @flow

import { SET_SEARCH_TERM } from './actions';

export const setSearchTerm = (searchTerm: string) => ({
  type: SET_SEARCH_TERM,
  payload: searchTerm
});
