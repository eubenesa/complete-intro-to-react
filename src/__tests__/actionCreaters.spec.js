// @flow

import moxios from 'moxios';
import { setSearchTerm, addApiData, getApiDetails } from '../actionCreaters';

describe('actionCreators', () => {
  it('setSearchTerm', () => {
    expect(setSearchTerm('homeland')).toMatchSnapshot();
  });

  const atlanta = {
    description: 'Two cousins, with different views on art versus commerce, on their way up through the Atlanta rap scene; "Earnest \'Earn\' Marks," an ambitious college drop-out and his estranged cousin, who suddenly becomes a star.',
    imdbID: 'tt4288182',
    poster: 'a.jpg',
    title: 'Atlanta',
    trailer: 'MpEdJ-mmTlY',
    year: '2008â€“2013',
    rating: '8.6'
  };

  it('addApiData', () => {
    expect(addApiData(atlanta)).toMatchSnapshot();
  });

  it('getApiDetails', (done: Function) => {
    const dispatchMock = jest.fn();
    moxios.withMock(() => {
      getApiDetails(atlanta.imdbID)(dispatchMock);
      moxios.wait(() => {
        const request = moxios.requests.mostRecent();
        request
          .respondWith({
            status: 200,
            response: atlanta
          })
          .then(() => {
            expect(request.url).toEqual(
              `http://localhost:3000/${atlanta.imdbID}`
            );
            expect(dispatchMock).toBeCalledWith(addApiData(atlanta));
            done();
          });
      });
    });
  });
});
