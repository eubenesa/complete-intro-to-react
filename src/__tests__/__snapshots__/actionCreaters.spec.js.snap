// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actionCreators addApiData 1`] = `
Object {
  "payload": Object {
    "description": "Two cousins, with different views on art versus commerce, on their way up through the Atlanta rap scene; \\"Earnest 'Earn' Marks,\\" an ambitious college drop-out and his estranged cousin, who suddenly becomes a star.",
    "imdbID": "tt4288182",
    "poster": "a.jpg",
    "rating": "8.6",
    "title": "Atlanta",
    "trailer": "MpEdJ-mmTlY",
    "year": "2008â€“2013",
  },
  "type": "ADD_API_DATA",
}
`;

exports[`actionCreators setSearchTerm 1`] = `
Object {
  "payload": "homeland",
  "type": "SET_SEARCH_TERM",
}
`;
